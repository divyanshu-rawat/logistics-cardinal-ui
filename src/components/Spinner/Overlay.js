import styled from 'styled-components';

const SpinnerOverlay = styled('div')`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background-color: ${({ theme }) => theme.spinnerOverlayBackgroundColor};
  z-index: ${({ theme }) => theme.zIndex.spinnerOverlay};
`;

export default SpinnerOverlay;
