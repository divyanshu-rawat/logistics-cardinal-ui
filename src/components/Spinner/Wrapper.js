import styled from 'styled-components';

const SpinnerWrapper = styled('div')`
  position: relative;
`;

export default SpinnerWrapper;
