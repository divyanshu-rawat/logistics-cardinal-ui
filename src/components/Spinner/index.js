import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Loader from 'react-md-spinner';
import { withTheme } from 'styled-components';

import SpinnerOverlay from './Overlay';
import SpinnerHolder from './Holder';
import SpinnerWrapper from './Wrapper';

const Spinner = ({ theme, size, children, spinWhen }) => (
  <SpinnerWrapper>
    {spinWhen && (
      <Fragment>
        <SpinnerOverlay />
        <SpinnerHolder>
          <Loader singleColor={theme.primaryColor} size={size} />
        </SpinnerHolder>
      </Fragment>
    )}
    {children}
  </SpinnerWrapper>
);

Spinner.propTypes = {
  theme: PropTypes.object.isRequired,
  size: PropTypes.number.isRequired,
  spinWhen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};

Spinner.defaultProps = {
  size: 50,
};

export default withTheme(Spinner);
