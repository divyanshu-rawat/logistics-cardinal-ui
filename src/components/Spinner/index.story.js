import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { boolean, number } from '@storybook/addon-knobs';

import withTests from '../../utils/withTests';

import { COMPONENTS } from '../../../.storybook/sections';

import Spinner from '.';
import Text from '../Text';

storiesOf(`${COMPONENTS}Spinner`, module)
  .addDecorator(
    withTests(
      'Spinner/tests/index',
      'Spinner/tests/Holder',
      'Spinner/tests/Overlay',
      'Spinner/tests/Wrapper',
    ),
  )
  .add(
    'Simple',
    withInfo()(() => (
      <Spinner spinWhen={boolean('Spin?', true)} size={number('Size', 50)}>
        <Text>
          Mussum Ipsum, cacilds vidis litro abertis. Mé faiz elementum girarzis,
          nisi eros vermeio. Cevadis im ampola pa arma uma pindureta. Si u mundo
          tá muito paradis? Toma um mé que o mundo vai girarzis! Quem num gosta
          di mé, boa gentis num é. Todo mundo vê os porris que eu tomo, mas
          ninguém vê os tombis que eu levo! Delegadis gente finis, bibendum
          egestas augue arcu ut est. Vehicula non. Ut sed ex eros. Vivamus sit
          amet nibh non tellus tristique interdum. Manduma pindureta quium dia
          nois paga. Suco de cevadiss deixa as pessoas mais interessantis.
          Mauris nec dolor in eros commodo tempor. Aenean aliquam molestie leo,
          vitae iaculis nisl. Nec orci ornare consequat. Praesent lacinia
          ultrices consectetur. Sed non ipsum felis. Diuretics paradis num copo
          é motivis de denguis. Atirei o pau no gatis, per gatis num morreus.
          Quem num gosta di mim que vai caçá sua turmis! Praesent malesuada urna
          nisi, quis volutpat erat hendrerit non. Nam vulputate dapibus.
          Casamentiss faiz malandris se pirulitá. Nullam volutpat risus nec leo
          commodo, ut interdum diam laoreet. Sed non consequat odio. Posuere
          libero varius. Nullam a nisl ut ante blandit hendrerit. Aenean sit
          amet nisi. Em pé sem cair, deitado sem dormir, sentado sem cochilar e
          fazendo pose. Interagi no mé, cursus quis, vehicula ac nisi. Tá
          deprimidis, eu conheço uma cachacis que pode alegrar sua vidis.
          Admodum accumsan disputationi eu sit. Vide electram sadipscing et per.
          Leite de capivaris, leite de mula manquis sem cabeça. Aenean aliquam
          molestie leo, vitae iaculis nisl. Não sou faixa preta cumpadi, sou
          preto inteiris, inteiris. A ordem dos tratores não altera o pão duris.
          Viva Forevis aptent taciti sociosqu ad litora torquent. Interessantiss
          quisso pudia ce receita de bolis, mais bolis eu num gostis. Suco de
          cevadiss, é um leite divinis, qui tem lupuliz, matis, aguis e
          fermentis. In elementis mé pra quem é amistosis quis leo. Copo furadis
          é disculpa de bebadis, arcu quam euismod magna. Si num tem leite então
          bota uma pinga aí cumpadi! Detraxit consequat et quo num tendi nada.
          Mais vale um bebadis conhecidiss, que um alcoolatra anonimis. Per
          aumento de cachacis, eu reclamis. Praesent vel viverra nisi. Mauris
          aliquet nunc non turpis scelerisque, eget. Pra lá , depois divoltis
          porris, paradis. Quem manda na minha terra sou euzis! Paisis, filhis,
          espiritis santis. Sapien in monti palavris qui num significa nadis i
          pareci latim.
        </Text>
      </Spinner>
    )),
  );
