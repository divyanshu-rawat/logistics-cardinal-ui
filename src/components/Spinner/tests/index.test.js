import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from '..';
import Theme from '../../../themes/rooster';

function renderComponent(size, spinWhen = false) {
  return shallow(
    <Spinner theme={Theme} size={size} spinWhen={spinWhen}>
      <span>some children</span>
    </Spinner>,
  ).dive();
}

describe('<Spinner />', () => {
  it('should render the component', () => {
    const tree = toJson(renderComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with different props', () => {
    const tree = toJson(renderComponent(100, true));
    expect(tree).toMatchSnapshot();
  });
});
