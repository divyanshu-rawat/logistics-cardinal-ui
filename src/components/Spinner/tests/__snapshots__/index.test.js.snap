// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render the component 1`] = `
<Wrapper__SpinnerWrapper>
  <span>
    some children
  </span>
</Wrapper__SpinnerWrapper>
`;

exports[`<Spinner /> should render the component with different props 1`] = `
<Wrapper__SpinnerWrapper>
  <Overlay__SpinnerOverlay />
  <Holder__SpinnerWrapper>
    <MDSpinner
      color1="rgb(66, 165, 245)"
      color2="rgb(239, 83, 80)"
      color3="rgb(253, 216, 53)"
      color4="rgb(76, 175, 80)"
      duration={1333}
      singleColor="#3f9bfd"
      size={100}
    />
  </Holder__SpinnerWrapper>
  <span>
    some children
  </span>
</Wrapper__SpinnerWrapper>
`;
